concurrent/actor/EventLoop.java

```diff
@@ -17,185 +17,190 @@
 
 package grakn.core.concurrent.actor;
 
+import grakn.core.common.exception.GraknException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
+import javax.annotation.concurrent.ThreadSafe;
 import java.util.PriorityQueue;
-import java.util.Random;
 import java.util.concurrent.LinkedTransferQueue;
 import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TransferQueue;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
+import static grakn.core.common.exception.ErrorMessage.Internal.ILLEGAL_OPERATION;
+import static grakn.core.common.exception.ErrorMessage.Internal.UNEXPECTED_INTERRUPTION;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@ThreadSafe
 public class EventLoop {
+
     private static final Logger LOG = LoggerFactory.getLogger(EventLoop.class);
     private static final Consumer<Throwable> DEFAULT_ERROR_HANDLER = e -> LOG.error("An unexpected error has occurred.", e);
 
-    private enum State {READY, RUNNING, STOPPED}
-
-    private State state;
-    private final TransferQueue<Job> jobs = new LinkedTransferQueue<>();
-    private final ScheduledJobQueue scheduledJobs = new ScheduledJobQueue();
+    private final TransferQueue<Job> submittedJobs;
+    private final ScheduledJobQueue scheduledJobs;
     private final Supplier<Long> clock;
-    private final Random random;
     private final Thread thread;
+    private State state;
 
-    public EventLoop(ThreadFactory threadFactory, Supplier<Long> clock, Random random) {
-        state = State.READY;
+    private enum State {READY, RUNNING, STOPPED}
+
+    public EventLoop(ThreadFactory threadFactory, Supplier<Long> clock) {
+        this.thread = threadFactory.newThread(this::loop);
         this.clock = clock;
-        this.random = random;
-        thread = threadFactory.newThread(this::loop);
+        submittedJobs = new LinkedTransferQueue<>();
+        scheduledJobs = new ScheduledJobQueue();
+        state = State.READY;
         thread.start();
     }
 
-    public void schedule(Runnable job, Consumer<Throwable> errorHandler) {
+    public void submit(Runnable runnable, Consumer<Throwable> errorHandler) {
         assert state != State.STOPPED : "unexpected state: " + state;
-
-        jobs.offer(new Job(job, errorHandler));
+        submittedJobs.offer(new Job(runnable, errorHandler));
     }
 
-    public EventLoop.Cancellable schedule(long deadline, Runnable job, Consumer<Throwable> errorHandler) {
+    public Cancellable schedule(Runnable runnable, Consumer<Throwable> errorHandler, long scheduleMillis) {
         assert state != State.STOPPED : "unexpected state: " + state;
-        return new Cancellable(deadline, job, errorHandler);
+        Cancellable cancellable = new Cancellable(runnable, errorHandler, scheduleMillis);
```

Comment:
Rework `Cancellable` API to be clearer.

The `schedule` and `cancel` method is executed through the event-loop. This is necessary in order to preserve determinism required by Grakn Cluster.